clients/unshaded/src/main/java/tachyon/client/block/RemoteBlockInStream.java

```diff
@@ -30,6 +33,13 @@
 public final class RemoteBlockInStream extends BufferedBlockInStream {
   /** The address of the worker to read the data from. */
   private final InetSocketAddress mLocation;
+  /** The returned lock id after acquiring the block lock. */
+  private final Long mLockId;
+
+  /** Client to communicate with the remote worker. */
+  private final WorkerClient mWorkerClient;
```

Comment:
Should there be a long-lived WorkerClient for the instream, or just acquire one on demand? I think maybe on-demand acquire/release might be less error prone. Maybe @calvinjia has thoughts on this?

Depending on if this is long-lived or on-demand, it may change how it is acquired or released in the other parts of this class.